src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,95 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { addComment } from '../services/comment';
+
+type Props = {
+  selectedPost: Post;
+  setComments: (value: Comment[] | { (prev: Comment[]): Comment[] }) => void,
+  setErrorMessage: (value: string) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setComments,
+  setErrorMessage,
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [comment, setComment] = useState('');
+  const [commentError, setCommentError] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
would be better to create 2 obj: 1 contains all form data, 2 contains all error data

Reply:
Thank you for your feedback! 
Just one thing to point: it says not the entire form should clear after submission, but only a comment input...
I can make it clear though.. 

'keep the name and email after the successful submit but clear a comment text;'